### PR TITLE
Add missing BM related function to platform_nodes

### PR DESCRIPTION
**Pull request by prsurve**
Fixes: #2745 
Fixes: #2746 

Signed-off-by: Pratik Surve <prsurve@redhat.com>

### CODE REVIEW

**Review by ebenahar — APPROVED** [review]

File: ocs_ci/ocs/platform_nodes.py [hunk 1]
```diff
@@ -1605,6 +1606,16 @@ def restart_nodes_teardown(self):
...
 
+    def restart_nodes_by_stop_and_start(self, nodes, force=True):
+        """
+        Restart Bare Metal Nodes
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            force (bool): True for force BM stop, False otherwise
+        """
+        self.baremetal.restart_baremetal_machines(nodes, force=force)
```
Comment (vasukulkarni) [inline ocs_ci/ocs/platform_nodes.py:1617]: why is this method same for both restart_nodes and restart_nodes_by_stop_and_start?

Comment (vasukulkarni) [reply ocs_ci/ocs/platform_nodes.py:1617]: in which case the function should be just an alias, one of the def is not needed.

Comment (clacroix12) [reply ocs_ci/ocs/platform_nodes.py:1617]: Yeah we could simply this by having an argument for `stop_and_start` and have one method handle both cases.

File: ocs_ci/ocs/platform_nodes.py [hunk 2]
```diff
@@ -1577,18 +1577,19 @@ def start_nodes(self, nodes, wait=True):
...
+    def restart_nodes(self, nodes, force=True, wait=True):
         """
         Restart Baremetal Machine
 
         Args:
             nodes (list): The OCS objects of the nodes
             force (bool): True for force BM stop, False otherwise
-
+            wait (bool): True if need to wait till the restarted OCP node
+                reaches READY state. False otherwise
```
Comment (vavuthu) [inline ocs_ci/ocs/platform_nodes.py:1588]: Add empty line below

File: ocs_ci/utility/baremetal.py [hunk 3]
```diff
@@ -195,18 +197,19 @@ def start_baremetal_machines(self, baremetal_machine, wait=True):
...
+                timeout=800
+            )
+            wait_for_nodes_status(
+                node_names=helpers.get_worker_nodes(),
+                status=constants.NODE_READY,
+                timeout=800
+            )
 
-    def restart_baremetal_machines(self, baremetal_machine, force=True):
+    def restart_baremetal_machines(self, baremetal_machine, force=True, wait=True):
```
Comment (yitzhak12) [inline ocs_ci/utility/baremetal.py:212]: Is it possible to add here also the `timeout` parameter?

File: ocs_ci/ocs/platform_nodes.py [hunk 4]
```diff
@@ -1577,18 +1577,19 @@ def start_nodes(self, nodes, wait=True):
         """
         self.baremetal.start_baremetal_machines(nodes, wait=wait)
 
-    def restart_nodes(self, nodes, force=True):
+    def restart_nodes(self, nodes, force=True, wait=True):
```
Comment (yitzhak12) [inline ocs_ci/ocs/platform_nodes.py:1580]: Is it possible to add here also the `timeout` parameter?

**Comment by petr-balogh** [comment]
@prsurve hey Pratik, are you going to finish with this PR? I see no activity here for long time. Thanks

**Pull request closed without merging**